src/soundiness.rkt

```diff
@@ -17,12 +17,9 @@
 
 (define (get-proof-errors proof pcontext ctx program-vars)
   (define proof-programs
-    (map (lambda (expr)
-           `(λ ,program-vars
-              ,(remove-rewrites expr)))
-         proof))
-  (define proof-errors
-    (map (lambda (x) (errors x pcontext ctx)) (remove-rewrites proof-programs)))
+    (for/list ([step (in-list proof)])
+      `(λ ,program-vars ,(remove-rewrites step))))
+  (define proof-errors (batch-errors proof-programs pcontext ctx))
```

Comment:
The switch to `batch-errors` is definitely good. I assume `remove-rewrites` isn't necessary because we already removed the rewrites from the body?